File: sdk/go/pulumi-language-go/testdata/extra-types/projects/l3-component-config-objects/primitiveComponent.go
package main

import (
	"fmt"

	"example.com/pulumi-primitive/sdk/go/v7/primitive"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

type PrimitiveComponentArgs struct {
	NumberArray pulumi.Float64ArrayInput
	BooleanMap  pulumi.BoolMapInput
}

type PrimitiveComponent struct {
	pulumi.ResourceState
}

func NewPrimitiveComponent(
	ctx *pulumi.Context,
	name string,
	args *PrimitiveComponentArgs,
	opts ...pulumi.ResourceOption,
) (*PrimitiveComponent, error) {
	var componentResource PrimitiveComponent
	err := ctx.RegisterComponentResource("components:index:PrimitiveComponent", name, &componentResource, opts...)
	if err != nil {
		return nil, err
	}
	_, err = primitive.NewResource(ctx, fmt.Sprintf("%s-res", name), &primitive.ResourceArgs{
		Boolean:     pulumi.Bool(true),
		Float:       pulumi.Float64(3.5),
		Integer:     pulumi.Int(3),
		String:      pulumi.String("plain"),
		NumberArray: args.NumberArray,
		BooleanMap:  args.BooleanMap,
	}, pulumi.Parent(&componentResource))
	if err != nil {
		return nil, err
	}
	err = ctx.RegisterResourceOutputs(&componentResource, pulumi.Map{})
	if err != nil {
		return nil, err
	}
	return &componentResource, nil
}

File: sdk/python/cmd/pulumi-language-python/testdata/setuppy/local/projects/l3-component-config-primitives/primitiveComponent.py
import pulumi
from pulumi import Input
from typing import Optional, Dict, TypedDict, Any
import pulumi_primitive as primitive

class PrimitiveComponentArgs(TypedDict, total=False):
    boolean: Input[bool]
    float: Input[float]
    integer: Input[int]
    string: Input[str]

class PrimitiveComponent(pulumi.ComponentResource):
    def __init__(self, name: str, args: PrimitiveComponentArgs, opts:Optional[pulumi.ResourceOptions] = None):
        super().__init__("components:index:PrimitiveComponent", name, args, opts)

        res = primitive.Resource(f"{name}-res",
            boolean=args["boolean"],
            float=args["float"],
            integer=args["integer"],
            string=args["string"],
            number_array=[
                float(-1),
                float(0),
                float(1),
            ],
            boolean_map={
                "t": True,
                "f": False,
            },
            opts = pulumi.ResourceOptions(parent=self))

        self.register_outputs({})
